toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -406,23 +437,15 @@ func (r *RpmRepoCloner) WhatProvides(pkgVer *pkgjson.PackageVer) (packageNames [
 	baseArgs := []string{
 		"provides",
 		provideQuery,
-		fmt.Sprintf("--disablerepo=%s", allRepoIDs),
 		releaseverCliArg,
 	}
 
-	effectiveCacheRepo := selectCorrectCacheRepoID()
-
 	// Consider the built (tooolchain, local) RPMs first, then the already cached, and finally all remote packages.
-	repoOrderList := []string{toolchainRepoId, builtRepoID, effectiveCacheRepo, allRepoIDs}
-	for _, repoID := range repoOrderList {
-		logger.Log.Debugf("Enabling repo ID: %s", repoID)
+	for _, reposArgs := range r.reposArgsList {
+		logger.Log.Debugf("Using repos args: %v", reposArgs)
 
 		err = r.chroot.Run(func() (err error) {
-			completeArgs := append(baseArgs, fmt.Sprintf("--enablerepo=%s", repoID))
-
-			if !r.usePreviewRepo {
-				completeArgs = append(completeArgs, fmt.Sprintf("--disablerepo=%s", previewRepoID))
-			}
+			completeArgs := append(baseArgs, reposArgs...)
```

Comment:
It looks like we never use more than one arg at a time here. It took me a sec to figure out what was going on. I'd recommend we switch this to just a `[]string` instead of `[][]string`.

```suggestion
			completeArgs := append(baseArgs, reposArgs)
```

Reply:
Per offline discussion, we do in fact want to keep a `[][]string` so we can preserve an inremental enabling of repos without recalculating the arguments.